src/SqlBinding/TriggerBinding/SqlTableWatcher.cs

```diff
@@ -0,0 +1,604 @@
+                }
+                _rows.Clear();
+            }
+            finally
+            {
+                // Want to do this before releasing the lock in case the renew leases timer goes off. It will see that
+                // the state is checking for changes and not renew the (just released) leases
+                // Only want to change the state if it was previously ProcessingChanges though. If it is stopped, for example,
+                // we don't want to start polling for changes again
+                Interlocked.CompareExchange(ref _state, CheckingForChanges, ProcessingChanges);
+                _renewLeasesLock.Release();
+            }
+        }
+
+        /// <summary>
+        /// Builds the query to create the worker table if one does not already exist (<see cref="CreateWorkerTableAsync"/>)
+        /// </summary>
+        /// <param name="connection">The connection to add to the returned SqlCommand</param>
+        /// <returns>The SqlCommand populated with the query and appropriate parameters</returns>
+        private async Task<SqlCommand> BuildCreateTableCommandAsync(SqlConnection connection)
+        {
+            await GetPrimaryKeysAsync();
+
+            var primaryKeysWithTypes = string.Empty;
+            var primaryKeysList = string.Empty;
+            foreach (var primaryKey in _primaryKeys.Keys)
+            {
+                string type;
+                _primaryKeys.TryGetValue(primaryKey, out type);
+                primaryKeysWithTypes += primaryKey + " " + type + ",\n";
+                primaryKeysList += primaryKey + ", ";
+            }
+            // Remove the trailing ", "
+            primaryKeysList = primaryKeysList.Substring(0, primaryKeysList.Length - 2);
+
+            var createTableString = 
+                $"IF OBJECT_ID(N\'{_workerTable}\', \'U\') IS NULL\n" +
+                $"BEGIN\n" +
+                $"CREATE TABLE {_workerTable} (\n" +
+                $"{primaryKeysWithTypes}" +
+                $"LeaseExpirationTime datetime2,\n" +
+                $"DequeueCount int,\n" +
+                $"VersionNumber bigint\n" +
+                $"PRIMARY KEY({primaryKeysList})\n" +
+                $");\n" +
+                $"END";
+            return new SqlCommand(createTableString, connection);
+        }
+
+        /// <summary>
+        /// Builds the query to check for changes on the user's table (<see cref="CheckForChangesAsync"/>)
+        /// </summary>
+        /// <param name="connection">The connection to add to the returned SqlCommand</param>
+        /// <param name="transaction">The transaction to add to the returned SqlCommand</param>
+        /// <returns>The SqlCommand populated with the query and appropriate parameters</returns>
+        private SqlCommand BuildCheckForChangesCommand(SqlConnection connection, SqlTransaction transaction)
+        {
+            string primaryKeysSelectList;
+            string primaryKeysInnerJoin;
```

Comment:
Tip: `StringBuilder` is the preferred method of building dynamic strings because it doesn't allocate multiple string objects like string concatenation does.